File: src/test/java/no/nav/syfo/mock/ArbeidsfordelingMock.java
package no.nav.syfo.mock;

import no.nav.tjeneste.virksomhet.arbeidsfordeling.v1.ArbeidsfordelingV1;
import no.nav.tjeneste.virksomhet.arbeidsfordeling.v1.FinnAlleBehandlendeEnheterListeUgyldigInput;
import no.nav.tjeneste.virksomhet.arbeidsfordeling.v1.FinnBehandlendeEnhetListeUgyldigInput;
import no.nav.tjeneste.virksomhet.arbeidsfordeling.v1.meldinger.WSFinnAlleBehandlendeEnheterListeRequest;
import no.nav.tjeneste.virksomhet.arbeidsfordeling.v1.meldinger.WSFinnAlleBehandlendeEnheterListeResponse;
import no.nav.tjeneste.virksomhet.arbeidsfordeling.v1.meldinger.WSFinnBehandlendeEnhetListeRequest;
import no.nav.tjeneste.virksomhet.arbeidsfordeling.v1.meldinger.WSFinnBehandlendeEnhetListeResponse;
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
import org.springframework.stereotype.Service;

@Service
@ConditionalOnProperty(value = "mockWS", havingValue = "true")
public class ArbeidsfordelingMock implements ArbeidsfordelingV1 {

    public WSFinnAlleBehandlendeEnheterListeResponse finnAlleBehandlendeEnheterListe(WSFinnAlleBehandlendeEnheterListeRequest wsFinnAlleBehandlendeEnheterListeRequest) throws FinnAlleBehandlendeEnheterListeUgyldigInput {
        throw new RuntimeException("Ikke implementert i mock");
    }

    public WSFinnBehandlendeEnhetListeResponse finnBehandlendeEnhetListe(WSFinnBehandlendeEnhetListeRequest wsFinnBehandlendeEnhetListeRequest) throws FinnBehandlendeEnhetListeUgyldigInput {
        throw new RuntimeException("Ikke implementert i mock");
    }

    public void ping() { }
}

File: src/test/kotlin/no/nav/syfo/mock/ArbeidsfordelingMock.kt
package no.nav.syfo.mock

import no.nav.tjeneste.virksomhet.arbeidsfordeling.v1.binding.FinnAlleBehandlendeEnheterListeUgyldigInput
import no.nav.tjeneste.virksomhet.arbeidsfordeling.v1.binding.FinnBehandlendeEnhetListeUgyldigInput
import no.nav.tjeneste.virksomhet.arbeidsfordeling.v1.binding.ArbeidsfordelingV1
import no.nav.tjeneste.virksomhet.arbeidsfordeling.v1.informasjon.Enhetsstatus
import no.nav.tjeneste.virksomhet.arbeidsfordeling.v1.informasjon.Organisasjonsenhet
import no.nav.tjeneste.virksomhet.arbeidsfordeling.v1.meldinger.FinnAlleBehandlendeEnheterListeRequest
import no.nav.tjeneste.virksomhet.arbeidsfordeling.v1.meldinger.FinnAlleBehandlendeEnheterListeResponse
import no.nav.tjeneste.virksomhet.arbeidsfordeling.v1.meldinger.FinnBehandlendeEnhetListeRequest
import no.nav.tjeneste.virksomhet.arbeidsfordeling.v1.meldinger.FinnBehandlendeEnhetListeResponse
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
import org.springframework.stereotype.Service

@Service
@ConditionalOnProperty(value = ["mockWS"], havingValue = "true")
class ArbeidsfordelingMock : ArbeidsfordelingV1 {

    @Throws(FinnAlleBehandlendeEnheterListeUgyldigInput::class)
    override fun finnAlleBehandlendeEnheterListe(
            wsFinnAlleBehandlendeEnheterListeRequest: FinnAlleBehandlendeEnheterListeRequest
    ): FinnAlleBehandlendeEnheterListeResponse {
        throw RuntimeException("Ikke implementert i mock")
    }

    @Throws(FinnBehandlendeEnhetListeUgyldigInput::class)
    override fun finnBehandlendeEnhetListe(
            wsFinnBehandlendeEnhetListeRequest: FinnBehandlendeEnhetListeRequest
    ): FinnBehandlendeEnhetListeResponse {
        val finnBehandlendeEnhetListeResponse = FinnBehandlendeEnhetListeResponse()
        val organisasjonsenhet = Organisasjonsenhet()
                .also {
                    it.enhetId = "enhet123"
                    it.status = Enhetsstatus.AKTIV
                }
        finnBehandlendeEnhetListeResponse.behandlendeEnhetListe.add(organisasjonsenhet)
        return finnBehandlendeEnhetListeResponse
    }

    override fun ping() {}
}
